receiver/netflowreceiver/README.md

```diff
@@ -0,0 +1,94 @@
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@evan-bradley](https://www.github.com/evan-bradley), [@dlopes7](https://www.github.com/dlopes7) |
+
+[development]: https://github.com/open-telemetry/opentelemetry-collector#development
+[contrib]: https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
+<!-- end autogenerated section -->
+
+The netflow receiver can listen for [netflow](https://en.wikipedia.org/wiki/NetFlow), [sflow](https://en.wikipedia.org/wiki/SFlow), and [ipfix](https://en.wikipedia.org/wiki/IP_Flow_Information_Export) data and convert it to OpenTelemetry logs. The receiver is based on the [goflow2](https://github.com/netsampler/goflow2) project.
+
+This gives OpenTelemetry users the capability of monitoring network traffic, and answer questions like:
+
+* Which protocols are passing through the network?
+* Which servers and clients are producing the highest amount of traffic?
+* What ports are involved in these network calls?
+* How many bytes and packets are being sent and received?
+
+## Getting started
+
+By default the receiver will listen for ipfix and netflow on port `2055` and sflow on port `6343`. The receiver can be configured to listen on different ports and protocols.
+
+Example configuration:
+
+```yaml
+receivers:
+  netflow:
+    - scheme: netflow
+      port: 2055
+      sockets: 16
+      workers: 32
+
+processors:
+  batch:
+    send_batch_size: 2000
+    timeout: 30s
+
+exporters:
+  debug:
+    verbosity: detailed
+
+service:
+  pipelines:
+    logs:
+      receivers: [netflow]
+      processors: [batch]
+      exporters: [debug]
+  telemetry:
+    logs:
+      level: debug
+```
+
+We recommend using the batch processor to reduce the number of log requests being sent to the exporter. The batch processor will batch log records together and send them in a single request to the exporter.
+
+You would then configure your network devices to send netflow, sflow, or ipfix data to the Collector on the specified ports.
+
+## Configuration
+
+| Field | Description | Examples | Default |
+|-------|-------------|--------| ------- |
+| scheme | The type of flow data that to receive | `sflow`, `netflow`, `flow` | `netflow` |
+| hostname | The hostname or IP address to bind to | `localhost` | `0.0.0.0` |
+| port | The port to bind to | `2055` | `2055` |
+| sockets | The number of sockets to use | 1 | 1 |
+| workers | The number of workers used to decode incoming flow messages | 2 | 2 |
+| queue_size | The size of the incoming netflow packets queue | 1000 | 1000000 |
+
+## Data format
+
+The netflow data is standardized for the different schemas and is converted to OpenTelemetry logs. The output will follow the format:
```

Comment:
Hello, and thank you for this initiative! 
I am wondering if the output should be adapted to something more alike the existing semantic conventions for otel? For instance, use `source.address` instead of `src_addr` ?

For what it's worth, the NetObserv tool (for kubernetes) is also able to write otel logs from network flows, but we've tried to use an output format that is close to the semantic convention: cf https://github.com/rhobs/observability-data-model/blob/main/network-observability.md#format-proposal
(with a lot of extra fields that are not in NetFlow spec)

I think it would be great if we could agree on a standardized output.

Reply:
goflow2 doc seems to suggest it's possible to provide a custom mapping with: `-mapping mapping.yaml`; the default (or example) file being here https://github.com/netsampler/goflow2/blob/026a16f92e771aaf017c46da0221d907957464b7/cmd/goflow2/mapping.yaml

Reply:
We are hoping to ditch the formatter to reduce overhead (decoding twice), and that file doesn't allow you to change the name of a field

But we can morph the structure of the message after we receive it, would need to define what the other fields should be called.

Still not clear if this is a redhat specific semantic or something that has been adopted by otel already

Reply:
sounds good
what about the fields that aren't in the current semantic conventions? Could they be added to the spec? (not sure who to ask, @crobert-1 maybe?)
There's the bytes/packets counters, the start/end timestamps, the sampler info...

Reply:
I would propose
`network.io.bytes`,`network.io.packets`, etc

`source.address`, `destination,address`

`source.port`, `destination.port`

`network.type`, `network.transport`, `network.protocol.name`

something akin to
```
     "network": {
       "type": "ipv6",
       "transport": "tcp",
       "protocol": {
         "name": "http",
         "number": 80
       },
       "flow": {
            "start": 1681583295157626000,
            "end": 1681583295157626000
       }
       "io": {
         "bytes": 1500,
         "packets": 1
       },
    },
    "source": {
        "address": "fd01::1",
        "port": 443
    },
    "destination": {
        "address": "fd01::2",
        "port": 50001
    }
```

Reply:
The format of the processed message is defined in a `.proto` file [here](https://github.com/netsampler/goflow2/blob/026a16f92e771aaf017c46da0221d907957464b7/pb/flow.proto) which then generates [this struct](https://github.com/netsampler/goflow2/blob/main/pb/flow.pb.go#L164)

I do agree that we should modify the message to follow a certain convention, but having it be multiple layers deep might not be feasible

I will look into writing a formatter instead of trying to modify the `FlowMessage` struct, this will incur some extra overhead but I think it is better in the long run

The mapping files don't help us here since we can't modify existing fields from the proto message

Reply:
(same here about the nested structure, I'd only expect flat key/values - but I'm not otlp maintainer :-) )